src/main/java/org/vietabroader/view/MainView.java

```diff
@@ -161,6 +163,16 @@ private void initUI() {
         c.insets = new Insets(0, 0, 0, 0);
         panelMain.add(createFooterPanel(), c);
 
+        this.addWindowListener(new WindowAdapter() {
+            @Override
+            public void windowClosing(WindowEvent e){
+                GoogleAPIUtils.signOut();
+                GlobalState currentState = GlobalState.getInstance();
+                currentState.setStatus(GlobalState.Status.SIGNED_OUT);
+                currentState.setUserEmail("");
```

Comment:
we exit after this so these last 3 lines are pretty unnecessary. Remove them and you can merge.